mypy/plugin.py

```diff
@@ -3,20 +3,39 @@
 from mypy.nodes import Expression, StrExpr, IntExpr, UnaryExpr, Context
 from mypy.types import (
     Type, Instance, CallableType, TypedDictType, UnionType, NoneTyp, FunctionLike, TypeVarType,
-    AnyType
+    AnyType, TypeList, UnboundType
 )
 from mypy.messages import MessageBuilder
 
 
 # Create an Instance given full name of class and type arguments.
 NamedInstanceCallback = Callable[[str, List[Type]], Type]
 
+AnalyzeArgListCallback = Callable[[TypeList], Optional[Tuple[List[Type],
+                                                             List[int],
+                                                             List[Optional[str]]]]]
+
 # Some objects and callbacks that plugins can use to get information from the
 # type checker or to report errors.
-PluginContext = NamedTuple('PluginContext', [('named_instance', NamedInstanceCallback),
-                                             ('msg', MessageBuilder),
-                                             ('context', Context)])
+PluginContext = NamedTuple(
+    'PluginContext',
+    [
+        ('named_instance', NamedInstanceCallback),
+        ('msg', MessageBuilder),
+        ('context', Context)
```

Comment:
Honestly I'm no fan of this spread-out way of formatting a NamedTuple, but if you do this, at least add a trailing comma.